apps/frontend/core-dapp/src/features/withdraw/WithdrawStore.ts

```diff
@@ -126,4 +151,47 @@ export class WithdrawStore {
 
     return this.root.ppoTokenStore.formatUnits(rewardBN)
   }
+
+  get withdrawLimit(): WithdrawLimit {
+    const {
+      globalAmountWithdrawnThisPeriod,
+      globalPeriodLength,
+      globalWithdrawLimitPerPeriod,
+      lastGlobalPeriodReset,
+    } = this.root.withdrawHookStore
+    const { nowInMs } = this.root.timerStore
+
+    if (globalPeriodLength === undefined || lastGlobalPeriodReset === undefined) {
+      return {
+        status: 'loading',
+      }
+    }
+
+    const periodAlreadyReset =
+      differenceInMilliseconds(nowInMs, lastGlobalPeriodReset) > globalPeriodLength
+
+    const limitInfo = getBalanceLimitInfo({
+      additionalAmount: this.withdrawalAmountBN,
+      cap: globalWithdrawLimitPerPeriod,
+      // If the reset window has passed, disregard the value of globalAmountWithdrawnThisPeriod.
+      // The amount withdrawn is effectively zero.
+      // When someone withdraws, globalAmountWithdrawnThisPeriod will update and thus the withdraw limit will be recomputed
+      currentAmount: periodAlreadyReset ? BigNumber.from(0) : globalAmountWithdrawnThisPeriod,
+      formatUnits: this.root.preCTTokenStore.formatUnits.bind(this.root.preCTTokenStore),
+    })
+
+    if (limitInfo.status === 'already-exceeded' || limitInfo.status === 'exceeded-after-transfer') {
+      const nextGlobalPeriodReset = addDuration(lastGlobalPeriodReset, globalPeriodLength)
+      const timeToReset = differenceInMilliseconds(nextGlobalPeriodReset, nowInMs) as DurationInMs
+
+      return {
+        ...limitInfo,
+        resetsIn: periodAlreadyReset ? undefined : timeToReset,
```

Comment:
if periodAlreadyReset is returned above with status = 'not-exceeded', do we still need this check here?

```suggestion
        resetsIn: timeToReset,
```

Reply:
As explained above, we can't return `not-exceeded` if `periodAlreadyReset`